keras/backend/theano_backend.py

```diff
@@ -642,6 +700,67 @@ def conv2d(x, kernel, strides=(1, 1), border_mode='valid', dim_ordering='th',
     return conv_out
 
 
+def conv3d(x, kernel, strides=(1, 1, 1), border_mode='valid', dim_ordering='th',
+           image_shape=None, filter_shape=None):
+    '''
+    Run on cuDNN if available.
```

Comment:
`conv3d2d` that is being used in this function uses `tensor.nnet.conv2d` for the convolutions. [This page](http://deeplearning.net/software/theano/library/tensor/nnet/conv.html) mentions that `nnet.conv2d` uses cuDNN by default if available, so I think that this function will indeed take advantage of cuDNN 2d convolution operation if available.

This makes me wonder why the conv2d function in this same file uses `dnn.dnn_conv` instead of `tensor.nnet.conv2d` when cuDNN is detected, instead of simply using `nnet.conv2d`.